src/lib/wsman-soap-envelope.c

```diff
@@ -661,6 +661,11 @@ int wsman_parse_credentials(WsXmlDocH doc, WsSubscribeInfo * subsInfo,
 		node = ws_xml_get_child(tnode, 0, XML_NS_TRUST, WST_TOKENTYPE);
 		value = ws_xml_get_node_text(node);
 		if(strcmp(value, WST_USERNAMETOKEN) == 0) {
+			if (subsInfo->username || subsInfo->password) {
```

Comment:
Uhm, this might break backwards compatibility. I wonder if this needs a `strict` mode configurable ?

Reply:
Btw, this is no "memleak" fix and should be done in a separate pr.

Reply:
It's really a memleak, but somewhat obscured. If we entering here twice the pointers to username and password maybe overwritten, generating memleak. I can instead release memory if pointers are non-null if you prefer. May be that fix will be lass intrusive...